src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java

```diff
@@ -108,6 +112,55 @@ else if (typeOfAst != TokenTypes.ANNOTATIONS) {
         }
     }
 
+    /**
+     * Extract full ident from dot ast.
+     *
+     * @param full the FullIdent to add to
+     * @param dotAst ast node of type {@link TokenTypes#DOT}
+     */
+    private static void extractFullIdentFromDotAst(FullIdent full, DetailAST dotAst) {
+        DetailAST firstChildToExtract = dotAst.getFirstChild();
+        while (firstChildToExtract.getType() == TokenTypes.LPAREN) {
+            firstChildToExtract = firstChildToExtract.getNextSibling();
+        }
```

Comment:
Are we able to replace this with `TokenUtil#findFirstTokenByPredicate`? There are a few cases here and at least one in `ChainedMethodCallWrapCheck.java`

Reply:
Good point, resolved.